servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -0,0 +1,336 @@
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * Manages the metadata of all blocks in managed space. This information is used by the
+ * TieredBlockStore, Allocator and Evictor.
+ * <p>
+ * This class is thread-safe and all operations on block metadata such as StorageTier, StorageDir
+ * should go through this class.
+ */
+public class BlockMetadataManager {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** A list of managed StorageTier */
+  private List<StorageTier> mTiers;
+  /** A map from tier alias to StorageTier */
+  private Map<Integer, StorageTier> mAliasToTiers;
+
+  private BlockMetadataManager() {}
+
+  private void initBlockMetadataManager(TachyonConf tachyonConf) throws IOException {
+    // Initialize storage tiers
+    int totalTiers = tachyonConf.getInt(Constants.WORKER_MAX_TIERED_STORAGE_LEVEL, 1);
+    mAliasToTiers = new HashMap<Integer, StorageTier>(totalTiers);
+    mTiers = new ArrayList<StorageTier>(totalTiers);
+    for (int level = 0; level < totalTiers; level ++) {
+      StorageTier tier = StorageTier.newStorageTier(tachyonConf, level);
+      mTiers.add(tier);
+      mAliasToTiers.put(tier.getTierAlias(), tier);
+    }
+  }
+
+  public static BlockMetadataManager newBlockMetadataManager(TachyonConf tachyonConf)
+      throws IOException {
+    BlockMetadataManager ret = new BlockMetadataManager();
+    ret.initBlockMetadataManager(tachyonConf);
+    return ret;
+  }
+
+  /**
+   * Gets the StorageTier given its tierAlias.
+   *
+   * @param tierAlias the alias of this tier
+   * @return the StorageTier object associated with the alias
+   * @throws IOException if tierAlias is not found
+   */
+  public synchronized StorageTier getTier(int tierAlias) throws IOException {
+    StorageTier tier = mAliasToTiers.get(tierAlias);
+    if (tier == null) {
+      throw new IOException("Cannot find tier with alias " + tierAlias);
+    }
+    return tier;
+  }
+
+  /**
+   * Gets the list of StorageTier managed.
+   *
+   * @return the list of StorageTiers
+   */
+  public synchronized List<StorageTier> getTiers() {
+    return mTiers;
+  }
+
+  /**
+   * Gets the list of StorageTier below the tier with the given tierAlias.
+   *
+   * @param tierAlias the alias of a tier
+   * @return the list of StorageTier
+   * @throws IOException if tierAlias is not found
+   */
+  public synchronized List<StorageTier> getTiersBelow(int tierAlias) throws IOException {
+    int level = getTier(tierAlias).getTierLevel();
+    return mTiers.subList(level + 1, mTiers.size());
+  }
+
+  /**
+   * Gets the amount of available space in given location in bytes.
+   *
+   * @param location location the check available bytes
+   * @return available bytes
+   */
+  public synchronized long getAvailableBytes(BlockStoreLocation location) throws IOException {
```

Comment:
We should probably clarify what this method should be used for. For example, the master may care about the total number of bytes available to the worker, whereas the allocator/evictor will care about the bytes at a storagedir granularity.